src/safe_mpc/controller.py

```diff
@@ -1,17 +1,256 @@
+import re
 import numpy as np
 import scipy.linalg as lin
-from .abstract import AbstractController
 from casadi import Function, norm_2
+from acados_template import AcadosOcp, AcadosOcpSolver
+from copy import deepcopy
+import casadi as cs
+import sympy as sym
+from .safe_set import NetSafeSet, AnalyticSafeSet
+from .cost_definition import *
+
+class AbstractController:
```

Comment:
Add a method "set_cost" that specifies the cost by passing an object of class AbstractCost.